src/desktop-header/DesktopHeader.jsx

```diff
@@ -0,0 +1,153 @@
+        >
+          <Avatar size="1.5em" src={avatar} alt="" className="mr-2" />
+          {username} <CaretIcon role="img" aria-hidden focusable="false" />
+        </MenuTrigger>
+        <MenuContent className="mb-0 dropdown-menu show dropdown-menu-right pin-right shadow py-2">
+          <DesktopUserMenuSlot menu={userMenu} />
+        </MenuContent>
+      </Menu>
+    );
+  }
+
+  renderLoggedOutItems() {
+    const { loggedOutItems } = this.props;
+    return <DesktopLoggedOutItemsSlot items={loggedOutItems} />;
+  }
+
+  render() {
+    const {
+      logo,
+      logoAltText,
+      logoDestination,
+      loggedIn,
+      intl,
+    } = this.props;
+    const logoProps = { src: logo, alt: logoAltText, href: logoDestination };
+    const logoClasses = getConfig().AUTHN_MINIMAL_HEADER ? 'mw-100' : null;
+
+    return (
+      <header className="site-header-desktop">
+        <a className="nav-skip sr-only sr-only-focusable" href="#main">{intl.formatMessage(messages['header.label.skip.nav'])}</a>
+        <div className={`container-fluid ${logoClasses}`}>
+          <div className="nav-container position-relative d-flex align-items-center">
+            <LogoSlot {...logoProps} />
+            <nav
+              aria-label={intl.formatMessage(messages['header.label.main.nav'])}
+              className="nav main-nav"
+            >
+              {this.renderMainMenu()}
+            </nav>
+            <nav
+              aria-label={intl.formatMessage(messages['header.label.secondary.nav'])}
+              className="nav secondary-menu-container align-items-center ml-auto"
+            >
+              {loggedIn
+                ? (
+                  <>
+                    {this.renderSecondaryMenu()}
+                    {this.renderUserMenu()}
+                  </>
+                ) : this.renderLoggedOutItems()}
+            </nav>
+          </div>
+        </div>
+      </header>
+    );
+  }
+}
+
+export const desktopHeaderDataShape = {
+  mainMenu: desktopHeaderMainOrSecondaryMenuDataShape,
+  secondaryMenu: desktopHeaderMainOrSecondaryMenuDataShape,
+  userMenu: desktopUserMenuDataShape,
+  loggedOutItems: desktopLoggedOutItemsDataShape,
+  logo: PropTypes.string,
+  logoAltText: PropTypes.string,
+  logoDestination: PropTypes.string,
+  avatar: PropTypes.string,
+  username: PropTypes.string,
+  loggedIn: PropTypes.bool,
+};
+
+DesktopHeader.propTypes = {
+  mainMenu: desktopHeaderDataShape.mainMenu,
+  secondaryMenu: desktopHeaderDataShape.secondaryMenumainMenu,
+  userMenu: desktopHeaderDataShape.userMenumainMenu,
+  loggedOutItems: desktopHeaderDataShape.loggedOutItemsmainMenu,
+  logo: desktopHeaderDataShape.logomainMenu,
+  logoAltText: desktopHeaderDataShape.logoAltTextmainMenu,
+  logoDestination: desktopHeaderDataShape.logoDestinationmainMenu,
+  avatar: desktopHeaderDataShape.avatarmainMenu,
+  username: desktopHeaderDataShape.usernamemainMenu,
+  loggedIn: desktopHeaderDataShape.loggedInmainMenu,
```

Comment:
I ended up using this pattern in a couple spots. At first I tried doing `...desktopHeaderDataShape` but I ran into "has no corresponding propTypes declaration" lint errors so I just did this to work around them. I'm more than open to any suggestions.

Reply:
It's either this or tacking on some linter exception (if there even is one).  Either way, I'm usually in favor of the more explicit solution, so... I'd keep it like you have it now.